app/views/teachers/EditStudentModal.coffee

```diff
@@ -34,6 +40,44 @@ module.exports = class EditStudentModal extends ModalView
     auth.sendRecoveryEmail(email).then =>
       @state.set { emailSent: true }
 
+  onClickRevokeStudentButton: (e) ->
+    button = $(e.currentTarget)
+    s = $.i18n.t('teacher.revoke_confirm').replace('{{student_name}}', @user.broadName())
+    return unless confirm(s)
+    prepaid = @user.makeCoursePrepaid()
+    button.text($.i18n.t('teacher.revoking'))
+    prepaid.revoke(@user, {
+      success: =>
+        @user.unset('coursePrepaid')
+        @prepaids.fetchMineAndShared().done(=> @render())
+      error: (prepaid, jqxhr) =>
+        msg = jqxhr.responseJSON.message
+        noty text: msg, layout: 'center', type: 'error', killer: true, timeout: 3000
+    })
+
+  # TODO: Same logic as in `TeacherClassView.coffee`
+  studentStatusString: (student) ->
+    status = student.prepaidStatus()
+    expires = student.get('coursePrepaid')?.endDate
+    string = switch status
+      when 'not-enrolled' then $.i18n.t('teacher.status_not_enrolled')
+      when 'enrolled' then (if expires then $.i18n.t('teacher.status_enrolled') else '-')
+      when 'expired' then $.i18n.t('teacher.status_expired')
+    return string.replace('{{date}}', moment(expires).utc().format('ll'))
+
+  onClickEnrollStudentButton: ->
+    return unless me.id is @classroom.get('ownerID')
+    prepaid = @prepaids.find((prepaid) -> prepaid.status() is 'available')
+    prepaid.redeem(@user, {
+      success: (prepaid) =>
+        @user.set('coursePrepaid', prepaid.pick('_id', 'startDate', 'endDate', 'type', 'includedCourseIDs'))
+      error: (prepaid, jqxhr) =>
+        msg = jqxhr.responseJSON.message
+        noty text: msg, layout: 'center', type: 'error', killer: true, timeout: 3000
+      complete: => 
+        @render()
```

Comment:
Add tracking of event defined in the jade file, line 22 and 26(data-event-action) like this: https://github.com/codecombat/codecombat/blob/master/app/views/courses/TeacherClassView.coffee#L422